src/languages/r.js

```diff
@@ -2,75 +2,154 @@
+                  }
+                ]
+              }
+            },
+            {
+              // Handle `@param` to highlight the parameter name following
+              // after.
+              className: 'doctag',
+              begin: '@param',
+              end: /$/,
+              contains: [
+                {
+                  className: 'variable',
+                  variants: [
+                    { begin: IDENT_RE },
+                    { begin: /`(?:\\.|[^`])+`/ }
+                  ],
+                  endsParent: true
+                }
+              ]
+            },
+            {
+              className: 'doctag',
+              begin: /@[a-zA-Z]+/
+            },
+            {
+              className: 'meta-keyword',
+              begin: /\\[a-zA-Z]+/,
+            }
+          ]
+        }
+      ),
+
       hljs.HASH_COMMENT_MODE,
+
       {
-        begin: IDENT_RE,
-        keywords: {
-          $pattern: IDENT_RE,
-          keyword:
-            'function if in break next repeat else for return switch while try tryCatch ' +
-            'stop warning require library attach detach source setMethod setGeneric ' +
-            'setGroupGeneric setClass ...',
-          literal:
-            'NULL NA TRUE FALSE T F Inf NaN NA_integer_|10 NA_real_|10 NA_character_|10 ' +
-            'NA_complex_|10'
-        },
-        relevance: 0
-      },
-      {
-        // hex value
-        className: 'number',
-        begin: "0[xX][0-9a-fA-F]+[Li]?\\b",
-        relevance: 0
-      },
-      {
-        // explicit integer
-        className: 'number',
-        begin: "\\d+(?:[eE][+\\-]?\\d*)?L\\b",
-        relevance: 0
-      },
-      {
-        // number with trailing decimal
-        className: 'number',
-        begin: "\\d+\\.(?!\\d)(?:i\\b)?",
-        relevance: 0
+        className: 'string',
+        contains: [hljs.BACKSLASH_ESCAPE],
+        variants: [
+          hljs.END_SAME_AS_BEGIN({ begin: /[rR]"(-*)\(/, end: /\)(-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /[rR]"(-*)\{/, end: /\}(-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /[rR]"(-*)\[/, end: /\](-*)"/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /[rR]'(-*)\(/, end: /\)(-*)'/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /[rR]'(-*)\{/, end: /\}(-*)'/ }),
+          hljs.END_SAME_AS_BEGIN({ begin: /[rR]'(-*)\[/, end: /\](-*)'/ }),
+          {begin: '"', end: '"', relevance: 0},
+          {begin: "'", end: "'", relevance: 0}
+        ],
       },
+
       {
-        // number
         className: 'number',
-        begin: "\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",
+        variants: [
+          // Special case: only hexadecimal binary powers can contain fractions.
+          { begin: /(?<![a-zA-Z0-9._])0[xX][0-9a-fA-F]+\.[0-9a-fA-F]*[pP][+-]?\d+i?/ },
+          { begin: /(?<![a-zA-Z0-9._])0[xX][0-9a-fA-F]+([pP][+-]?\d+)?[Li]?/ },
+          { begin: /(?<![a-zA-Z0-9._])(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?[Li]?/ }
```

Comment:
Sorry I'm feeling stupid, remind me what edge case we're guarding against here that `\b` doesn't cover?  Is it just `_` or?

Reply:
Essentially it’s just `.`, since `_` is part of `\w` and thus makes a word boundary, isn’t it? The purpose is to prevent matching in identifiers, which allow all of the characters in the negative lookbehind.